### PR TITLE
chore: Document `PlatformDispatcher.onError` usage

### PR DESCRIPTION
**Pull request by denrase**
#skip-changelog

## :scroll: Description

Document `PlatformDispatcher.onError` usage

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: dart/lib/src/sentry.dart [hunk 1]
```diff
@@ -33,6 +33,9 @@ class Sentry {
   /// Initializes the SDK
   /// passing a [AppRunner] callback allows to run the app within its own error
   /// zone ([`runZonedGuarded`](https://api.dart.dev/stable/2.10.4/dart-async/runZonedGuarded.html))
+  /// Starting with Flutter 3.3, the the SDK will record errors using the
+  /// [`onError`](https://api.flutter.dev/flutter/dart-ui/PlatformDispatcher/onError.html) callback
+  /// instead of running in an own error zone.
```
Comment (marandaneto) [inline dart/lib/src/sentry.dart]: I'd rephrase this, looks like it's doing both things now, or add the version constraint for the line above as well.

Comment (denrase) [reply dart/lib/src/sentry.dart]: I removed this, since this is behaviour for the flutter sdk. The dart sdk on its own will still run in `runZonedGuarded` as a default.

File: flutter/README.md [hunk 2]
```diff
@@ -38,7 +42,31 @@ Future<void> main() async {
...
+  await SentryFlutter.init(
+    (options) {
+      options.dsn = 'https://example@sentry.io/add-your-dsn-here';
+    },
+  );
+
+  PlatformDispatcher.instance.onError = (Object exception, StackTrace stackTrace) async {
+    // Your code here...
+    await Sentry.captureException(exception, stackTrace: stackTrace);
+  };
```
Comment (marandaneto) [inline flutter/README.md]: Why do we need that? `SentryFlutter` should install the `onError` automatically.

File: flutter/README.md [hunk 3]
```diff
@@ -122,7 +151,8 @@ Or [try out the Alpha version of the Sentry Dart Plugin](https://github.com/gets
 
 - Use a `try/catch` block.
 - Use a `catchError` block for `Futures`, examples on [dart.dev](https://dart.dev/guides/libraries/futures-error-handling).
-- The SDK already runs your `callback` on an error handler, e.g. using [runZonedGuarded](https://api.flutter.dev/flutter/dart-async/runZonedGuarded.html), events caught by the `runZonedGuarded` are captured automatically.
+- Starting with Flutter 3.3 the SDK uses [`PlatformDispatcher.onError`](https://docs.flutter.dev/testing/errors#errors-not-caught-by-flutter) to capture errors automatically.
+- Prior to Flutter 3.3 the SDK runs `appRunner` callback inside [`runZonedGuarded`](https://api.flutter.dev/flutter/dart-async/runZonedGuarded.html) to capture errors automatically.
```
Comment (marandaneto) [inline flutter/README.md]:
```suggestion
- Prior to Flutter 3.3 the SDK runs the `appRunner` callback inside the [`runZonedGuarded`](https://api.flutter.dev/flutter/dart-async/runZonedGuarded.html) to capture errors automatically.
```

File: flutter/README.md [hunk 4]
```diff
@@ -23,6 +23,10 @@ It will capture errors in the native layer, including (Java/Kotlin/C/C++ for And
 
 - Initialize the Sentry SDK using the DSN issued by Sentry.io:
 
+- Errors not caught by flutter will be reported automatically by the SDK
```
Comment (marandaneto) [inline flutter/README.md]:
```suggestion
- Errors not caught by Flutter will be reported automatically by the SDK
```

File: flutter/README.md [hunk 5]
```diff
@@ -23,6 +23,10 @@ It will capture errors in the native layer, including (Java/Kotlin/C/C++ for And
...
+  - Starting with Flutter 3.3, the SDK uses [`PlatformDispatcher.onError`](https://docs.flutter.dev/testing/errors#errors-not-caught-by-flutter) to report errors not caught by flutter.
```
Comment (marandaneto) [inline flutter/README.md]: > to report errors not caught by flutter.

That's confusing, Flutter is the one that offers the callback, the developers/users are the ones not doing the try/catch

File: flutter/README.md (hunk 2 above)
Comment (denrase) [reply flutter/README.md]: Just a sample for users if they want to use the `onError` callback on their own, for whatever reason, they still need to call through to sentry.

**Comment by denrase** [comment]
@marandaneto Removed the extra text from the tips section, as it only duplicates the one from above.

File: flutter/README.md (hunk 2 above)
Comment (marandaneto) [reply flutter/README.md]: This is not needed since `PlatformDispatcher.instance.onError` is installed automatically.
the run zoned guard was needed because it was a Flutter limitation.

**Comment by marandaneto** [comment]
> @marandaneto Removed the extra text from the tips section, as it only duplicates the one from above.

Still missing https://github.com/getsentry/sentry-dart/pull/1082#discussion_r1004442446

File: flutter/README.md (hunk 2 above)
Comment (vlazdra) [reply flutter/README.md]: I stumbled upon this PR, while looking for exactly this piece of information.
Indeed, there is a need to allow the developer to implement the PlatformDispatcher on our own, and handle some custom logic. But since it's a single function, not a list of handlers, if we both setup the handler, we might cancel each other out.
So there needs to be a way of passing along the error to Sentry, once the custom logic has been completed in a way. Exactly like the example mentioned here.
Now, the only thing I'm personally uncertain is if this would affect Sentry in any way, of not setting the Handler on it's own.

Comment (ueman) [reply flutter/README.md]: Sentry doesn't override the handler but wraps it. See https://github.com/getsentry/sentry-dart/blob/main/flutter/lib/src/integrations/on_error_integration.dart

It's only overridden if the user sets it after initializing Sentry.

Comment (vlazdra) [reply flutter/README.md]: Ahhh I see, the 44th line explains a lot.

So you are actually keeping an instance of whatever was setup as the handler, before Sentry integrates it's own. And then whenever the system issues an `onError` call, you can call the handler that was originally setup, and then handle it on your own.

Thanks for the headsup @ueman , really appreciate it! 👍

**Review by marandaneto — APPROVED** [review]
